# Analyse de la Qualité du Vin

## 1. Contexte et Problématique

### Domaine métier
Ce projet porte sur l'analyse de la qualité du vin, spécifiquement les vins portugais "Vinho Verde" (vin vert).

### Problématique
- **Objectif principal** : Modéliser les préférences du vin en utilisant les propriétés physicochimiques
- **Approche** : Régression et classification basées sur les données d'analyse sensorielle
- **Enjeu** : Prédire la qualité d'un vin à partir de ses caractéristiques physico-chimiques (sans information sur la variété de raisin, la marque, le prix, etc.)

### Origine des données
- **Source** : Dataset public de recherche (Cortez et al., 2009)
- **Création** : Paulo Cortez et al. - Université du Minho (Braga, Portugal) en 2009
- **Publication** : Decision Support Systems, Elsevier

### Variantes disponibles
- Vin rouge ("Vinho Verde" rouge)
- Vin blanc ("Vinho Verde" blanc)

In [1]:
import pandas as pd

df = pd.read_csv("../data/winequality-red.csv", sep=";")
df.head()


fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0            7.4              0.70         0.00             1.9      0.076   
1            7.8              0.88         0.00             2.6      0.098   
2            7.8              0.76         0.04             2.3      0.092   
3           11.2              0.28         0.56             1.9      0.075   
4            7.4              0.70         0.00             1.9      0.076   

   free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0                 11.0                  34.0   0.9978  3.51       0.56   
1                 25.0                  67.0   0.9968  3.20       0.68   
2                 15.0                  54.0   0.9970  3.26       0.65   
3                 17.0                  60.0   0.9980  3.16       0.58   
4                 11.0                  34.0   0.9978  3.51       0.56   

   alcohol  quality  
0      9.4        5  
1      9.8        5  
2      9.8        5  
3      9.8        6  
4      9.4        5

## 2. Compréhension des Données

### 2.1 Structure générale
Les données sont organisées dans **2 fichiers CSV distincts** (séparés par des points-virgules):

In [2]:
import pandas as pd
import numpy as np

# Chargement des données
df_red = pd.read_csv("../data/winequality-red.csv", sep=";")
df_white = pd.read_csv("../data/winequality-white.csv", sep=";")

print("=" * 80)
print("DONNÉES - VIN ROUGE")
print("=" * 80)
print(f"Dimensions : {df_red.shape[0]} observations × {df_red.shape[1]} variables")
print(f"\nPremières lignes :")
print(df_red.head())

print("\n" + "=" * 80)
print("DONNÉES - VIN BLANC")
print("=" * 80)
print(f"Dimensions : {df_white.shape[0]} observations × {df_white.shape[1]} variables")
print(f"\nPremières lignes :")
print(df_white.head())

DONNÉES - VIN ROUGE
Dimensions : 1599 observations × 12 variables

Premières lignes :
   fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0            7.4              0.70         0.00             1.9      0.076   
1            7.8              0.88         0.00             2.6      0.098   
2            7.8              0.76         0.04             2.3      0.092   
3           11.2              0.28         0.56             1.9      0.075   
4            7.4              0.70         0.00             1.9      0.076   

   free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0                 11.0                  34.0   0.9978  3.51       0.56   
1                 25.0                  67.0   0.9968  3.20       0.68   
2                 15.0                  54.0   0.9970  3.26       0.65   
3                 17.0                  60.0   0.9980  3.16       0.58   
4                 11.0                  34.0   0.9978  3.51       0.56   



### 2.2 Variables et Descriptions

In [3]:
# Noms et descriptions des variables
variables_description = {
    "fixed acidity": "Acidité fixe (acides qui ne s'évaporent pas facilement)",
    "volatile acidity": "Acidité volatile (acides responsables du goût de vinaigre)",
    "citric acid": "Acide citrique (ajoute de la fraîcheur et du goût)",
    "residual sugar": "Sucre résiduel (sucre restant après fermentation)",
    "chlorides": "Chlorures (sel dans le vin)",
    "free sulfur dioxide": "Dioxyde de soufre libre (agent conservateur)",
    "total sulfur dioxide": "Dioxyde de soufre total",
    "density": "Densité (poids du vin par unité de volume)",
    "pH": "Acidité (échelle 0-14, < 7 = acide)",
    "sulphates": "Sulfates (additif antimicrobien)",
    "alcohol": "Alcool (% d'alcool)",
    "quality": "Qualité (score 0-10, basé sur évaluation sensorielle)"
}

print("=" * 80)
print("VARIABLES DU DATASET")
print("=" * 80)
for i, (var, desc) in enumerate(variables_description.items(), 1):
    print(f"{i:2d}. {var:25s} → {desc}")

print("\n" + "=" * 80)
print("CLASSIFICATION DES VARIABLES")
print("=" * 80)
print("\n📊 VARIABLES D'ENTRÉE (Features) - 11 variables physicochimiques:")
print("   • Acidité : fixed acidity, volatile acidity, citric acid, pH")
print("   • Sucres : residual sugar")
print("   • Minéraux : chlorides, sulphates")
print("   • Conservateurs : free sulfur dioxide, total sulfur dioxide")
print("   • Propriétés physiques : density, alcohol")
print("\n🎯 VARIABLE DE SORTIE (Target) - 1 variable sensorielle:")
print("   • quality : Score de qualité (0 = très mauvais, 10 = excellent)")

VARIABLES DU DATASET
 1. fixed acidity             → Acidité fixe (acides qui ne s'évaporent pas facilement)
 2. volatile acidity          → Acidité volatile (acides responsables du goût de vinaigre)
 3. citric acid               → Acide citrique (ajoute de la fraîcheur et du goût)
 4. residual sugar            → Sucre résiduel (sucre restant après fermentation)
 5. chlorides                 → Chlorures (sel dans le vin)
 6. free sulfur dioxide       → Dioxyde de soufre libre (agent conservateur)
 7. total sulfur dioxide      → Dioxyde de soufre total
 8. density                   → Densité (poids du vin par unité de volume)
 9. pH                        → Acidité (échelle 0-14, < 7 = acide)
10. sulphates                 → Sulfates (additif antimicrobien)
11. alcohol                   → Alcool (% d'alcool)
12. quality                   → Qualité (score 0-10, basé sur évaluation sensorielle)

CLASSIFICATION DES VARIABLES

📊 VARIABLES D'ENTRÉE (Features) - 11 variables physicochimiques:


### 2.3 Types de Données et Caractéristiques

In [4]:
print("=" * 80)
print("TYPES DE DONNÉES")
print("=" * 80)
print("\nVin Rouge:")
print(df_red.dtypes)
print("\nVin Blanc:")
print(df_white.dtypes)

print("\n" + "=" * 80)
print("STATISTIQUES DESCRIPTIVES - VIN ROUGE")
print("=" * 80)
print(df_red.describe().T)

print("\n" + "=" * 80)
print("STATISTIQUES DESCRIPTIVES - VIN BLANC")
print("=" * 80)
print(df_white.describe().T)

TYPES DE DONNÉES

Vin Rouge:
fixed acidity           float64
volatile acidity        float64
citric acid             float64
residual sugar          float64
chlorides               float64
free sulfur dioxide     float64
total sulfur dioxide    float64
density                 float64
pH                      float64
sulphates               float64
alcohol                 float64
quality                   int64
dtype: object

Vin Blanc:
fixed acidity           float64
volatile acidity        float64
citric acid             float64
residual sugar          float64
chlorides               float64
free sulfur dioxide     float64
total sulfur dioxide    float64
density                 float64
pH                      float64
sulphates               float64
alcohol                 float64
quality                   int64
dtype: object

STATISTIQUES DESCRIPTIVES - VIN ROUGE
                       count       mean        std      min      25%  \
fixed acidity         1599.0   8.319637   1.741096  

## 3. Mode de Collecte des Données

### 3.1 Processus de collecte
- **Variables d'entrée** : Tests **objectifs et mesurables** en laboratoire (physicochimiques)
- **Variable de sortie** : Données **sensorielles** (qualité basée sur dégustation)
  - Chaque vin a été évalué par **au moins 3 experts** en dégustation
  - Score final = **médiane des évaluations** des experts
  - Échelle : 0 (très mauvais) à 10 (excellent)

### 3.2 Limitations et particularités
- ✅ **Données complètes** : Aucune valeur manquante
- ❌ **Données partielles** : Informations limitées pour préserver la confidentialité
  - Pas de données sur : variété de raisins, marque, prix de vente, région précise, etc.
- ⚠️ **Données déséquilibrées** : 
  - Beaucoup plus de vins « normaux » que de vins « excellents » ou « très mauvais »
  - Distribution des scores de qualité est asymétrique

### 3.3 Potentiel du dataset
- 🔬 Testé avec succès en machine learning (SVM donne les meilleurs résultats)
- 📊 Peut être traité comme un problème de **régression** (prédire la note exacte) ou de **classification** (prédire une catégorie)
- 🎯 Bon candidat pour : 
  - Sélection de variables (feature selection) - certaines variables sont corrélées
  - Détection d'anomalies (identifier les rares vins excellents ou mauvais)

In [5]:
print("=" * 80)
print("ANALYSE DÉTAILLÉE DE LA QUALITÉ")
print("=" * 80)

print("\n📌 VIN ROUGE - Distribution de la Qualité:")
print(df_red['quality'].value_counts().sort_index())
print(f"\nMoyenne : {df_red['quality'].mean():.2f} | Médiane : {df_red['quality'].median():.0f} | Mode : {df_red['quality'].mode()[0]:.0f}")

print("\n📌 VIN BLANC - Distribution de la Qualité:")
print(df_white['quality'].value_counts().sort_index())
print(f"\nMoyenne : {df_white['quality'].mean():.2f} | Médiane : {df_white['quality'].median():.0f} | Mode : {df_white['quality'].mode()[0]:.0f}")

print("\n" + "=" * 80)
print("DÉTECTION DE VALEURS MANQUANTES")
print("=" * 80)
print(f"\nVin Rouge - Valeurs manquantes : {df_red.isnull().sum().sum()} (Confirmé : ✅ Pas de données manquantes)")
print(f"Vin Blanc - Valeurs manquantes : {df_white.isnull().sum().sum()} (Confirmé : ✅ Pas de données manquantes)")

print("\n" + "=" * 80)
print("COMPARAISON ROUGE vs BLANC")
print("=" * 80)
comparison_data = {
    "Métrique": ["Nb observations", "Alcool (moy %)", "Densité (moy)", "Acidité pH (moy)", "Qualité (moy)"],
    "Vin Rouge": [
        f"{df_red.shape[0]}",
        f"{df_red['alcohol'].mean():.2f}",
        f"{df_red['density'].mean():.5f}",
        f"{df_red['pH'].mean():.2f}",
        f"{df_red['quality'].mean():.2f}"
    ],
    "Vin Blanc": [
        f"{df_white.shape[0]}",
        f"{df_white['alcohol'].mean():.2f}",
        f"{df_white['density'].mean():.5f}",
        f"{df_white['pH'].mean():.2f}",
        f"{df_white['quality'].mean():.2f}"
    ]
}
comparison_df = pd.DataFrame(comparison_data)
print(comparison_df.to_string(index=False))

ANALYSE DÉTAILLÉE DE LA QUALITÉ

📌 VIN ROUGE - Distribution de la Qualité:
quality
3     10
4     53
5    681
6    638
7    199
8     18
Name: count, dtype: int64

Moyenne : 5.64 | Médiane : 6 | Mode : 5

📌 VIN BLANC - Distribution de la Qualité:
quality
3      20
4     163
5    1457
6    2198
7     880
8     175
9       5
Name: count, dtype: int64

Moyenne : 5.88 | Médiane : 6 | Mode : 6

DÉTECTION DE VALEURS MANQUANTES

Vin Rouge - Valeurs manquantes : 0 (Confirmé : ✅ Pas de données manquantes)
Vin Blanc - Valeurs manquantes : 0 (Confirmé : ✅ Pas de données manquantes)

COMPARAISON ROUGE vs BLANC
        Métrique Vin Rouge Vin Blanc
 Nb observations      1599      4898
  Alcool (moy %)     10.42     10.51
   Densité (moy)   0.99675   0.99403
Acidité pH (moy)      3.31      3.19
   Qualité (moy)      5.64      5.88


## 4. Résumé Exécutif

| Aspect | Détails |
|--------|---------|
| **Domaine** | Analyse sensorielle et prédiction de qualité de vin |
| **Datasets** | 2 datasets : vin rouge (1599 obs) + vin blanc (4898 obs) = **6497 observations** |
| **Variables** | 11 variables d'entrée + 1 variable cible = **12 variables totales** |
| **Types** | Toutes les variables sont **numériques (float/int)** |
| **Données manquantes** | ✅ **Aucune** |
| **Collecte** | Tests physicochimiques en labo + Évaluations sensorielles (≥3 experts) |
| **Cible (Quality)** | Score ordinal 0-10 basé sur consensus d'experts |
| **Déséquilibre** | Peu de vins excellents (9-10) et mauvais (3 ou moins) |
| **Format** | CSV séparé par **point-virgule (;)** |
| **Qualité données** | ⭐⭐⭐⭐⭐ Excellente (publique, citée 1000+ fois, bien documentée) |